executor/set_test.go

```diff
@@ -123,6 +123,12 @@ func (s *testSuite) TestSetVar(c *C) {
 	c.Assert(vars.SkipConstraintCheck, IsTrue)
 	tk.MustExec("set @@tidb_skip_constraint_check = '0'")
 	c.Assert(vars.SkipConstraintCheck, IsFalse)
+
+	tk.MustExec("set global avoid_temporal_upgrade = on")
```

Comment:
true or false have been handled in parser.y in Literal.
This issue only handle the syntax error produced by SET variable = ON